base/untar.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+#
+# Download and extract tarball archive from the network.
+
+set -ex
+
+case "$1" in
+    -r) dest=/ ;;
+    -l) dest=. ;;
+    *) >&2 echo "Invalid extraction mode: must be either root (-r) or local (-l)."
+       exit 1;
+    ;;
+esac
+
+shift
+
+urls=$@
+
+for url in $urls; do
+    download_dir=$(mktemp -d)
+    wget $url -P $download_dir
+    tar --no-same-owner -xvf $download_dir/* -C $dest
+    rm -rf $download_dir
+done
+
+ldconfig
```

Comment:
This should only be done in places where we know libraries are being installed, after running the script.